libcaf_core/caf/flow/byte.test.cpp

```diff
@@ -0,0 +1,220 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/flow/byte.hpp"
+
+#include "caf/test/caf_test_main.hpp"
+#include "caf/test/fixture/deterministic.hpp"
+#include "caf/test/fixture/flow.hpp"
+#include "caf/test/scenario.hpp"
+#include "caf/test/test.hpp"
+
+#include "caf/flow/multicaster.hpp"
+
+using namespace caf;
+
+namespace {
+
+/// Compares two chunks for equality.
+[[maybe_unused]] bool operator==(const caf::chunk& lhs, const caf::chunk& rhs) {
+  return std::equal(lhs.bytes().begin(), lhs.bytes().end(),
+                    rhs.bytes().begin());
+}
+
+template <class T>
+auto convert_to_byte_arr(const T& data) {
+  std::vector<std::byte> byte_arr;
+  std::transform(data.begin(), data.end(), std::back_inserter(byte_arr),
+                 [](auto element) { return static_cast<std::byte>(element); });
+  return byte_arr;
+}
+
+WITH_FIXTURE(test::fixture::flow) {
+
+SCENARIO("to_chunks can create chunks from a byte array source") {
+  GIVEN("a container") {
```

Comment:
The container is only for building an `observable`:

```suggestion
  GIVEN("an observable<byte>") {
```

(same for the other `GIVEN`)